src/app/qgsmaptoolselectutils.cpp

```diff
@@ -243,6 +243,8 @@ QgsFeatureIds QgsMapToolSelectUtils::getMatchingFeatures( QgsMapCanvas *canvas,
   {
     // a zero width buffer is safer than calling make valid here!
     selectGeomTrans = selectGeomTrans.buffer( 0, 1 );
+    if ( selectGeomTrans.isNull() || selectGeomTrans.isEmpty() )
```

Comment:
```suggestion
    if ( selectGeomTrans.isEmpty() )
```
(the isEmpty check will return true for null geometries too!)